modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -278,9 +277,8 @@ private DeployTask createNewDeployTask(DeployRequest deployRequest) {
                 || Objects.isNull(existingServiceTemplate.getOcl())) {
             throw new ServiceTemplateNotRegistered("No available service templates found.");
         }
-        if (ServiceRegistrationState.APPROVED
-                != existingServiceTemplate.getServiceRegistrationState()) {
-            String errMsg = String.format("Found service template with id %s but not approved.",
+        if (Objects.equals(existingServiceTemplate.getAvailableInCatalog(), false)) {
+            String errMsg = String.format("Found service template with id %s but is unavailable.",
```

Comment:
for this case, we will need some other exception type. because 'ServiceTemplateNotApproved' would not be valid.